src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -0,0 +1,359 @@
+            buttonBuild.setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.build"));
+        }
+        else if(building.getBuildingLevel() == building.getBuildingMaxLevel())
+        {
+            buttonBuild.setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.switchStyle"));
+        }
+        else
+        {
+            buttonBuild.setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.upgrade"));
+        }
+    }
+
+    /**
+     * On cancel button.
+     */
+    private void cancelClicked()
+    {
+        building.openGui(false);
+    }
+
+    /**
+     * On confirm button.
+     */
+    private void confirmClicked()
+    {
+        MineColonies.getNetwork().sendToServer(new BuildingSetStyleMessage(building, styles.get(stylesDropDownList.getSelectedIndex())));
+        if(building.getBuildingLevel() == building.getBuildingMaxLevel())
+        {
+            MineColonies.getNetwork().sendToServer(new BuildRequestMessage(building, BuildRequestMessage.REPAIR));
+        }
+        else
+        {
+            MineColonies.getNetwork().sendToServer(new BuildRequestMessage(building, BuildRequestMessage.BUILD));
+        }
+        cancelClicked();
+    }
+
+    /**
+     * Update the styles list but try to keep the same one.
+     */
+    private void updateStyles()
+    {
+        styles = Structures.getStylesFor(building.getSchematicName());
+        int newIndex = styles.indexOf(building.getStyle());
+        if (newIndex == -1)
+        {
+            newIndex = 0;
+        }
+
+        final boolean enabled;
+        if(Settings.instance.isStaticSchematicMode())
+        {
+            enabled = false;
+        }
+        else
+        {
+            enabled = styles.size() > 1;
+        }
+
+        findPaneOfTypeByID(BUTTON_PREVIOUS_STYLE_ID, Button.class).setEnabled(enabled);
+        findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class).setEnabled(enabled);
+        findPaneOfTypeByID(BUTTON_NEXT_STYLE_ID, Button.class).setEnabled(enabled);
+        stylesDropDownList.setSelectedIndex(newIndex);
+    }
+
+    /**
+     * Clears and resets/updates all resources.
+     */
+    private void updateResources()
+    {
+        final World world = Minecraft.getMinecraft().world;
+        resources.clear();
+
+        final int nextLevel = building.getBuildingLevel() == building.getBuildingMaxLevel() ? building.getBuildingMaxLevel() : building.getBuildingLevel() + 1;
+        final StructureName sn = new StructureName(Structures.SCHEMATICS_PREFIX, styles.get(stylesDropDownList.getSelectedIndex()) ,
+                building.getSchematicName() + nextLevel);
+        final StructureWrapper wrapper = new StructureWrapper(world, sn.toString());
+        wrapper.setPosition(building.getLocation());
+        wrapper.rotate(building.getRotation(), world, building.getLocation(), building.isMirrored() ? Mirror.FRONT_BACK : Mirror.NONE);
+        while (wrapper.findNextBlock())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)